internal/dag/edge.go

```diff
@@ -25,7 +21,7 @@ type basicEdge struct {
 }
 
 func (e *basicEdge) Hashcode() interface{} {
-	return fmt.Sprintf("%p-%p", e.S, e.T)
+	return [...]interface{}{e.S, e.T}
```

Comment:
Slightly more efficient might be `return *e`, since the struct as the same memory layout as a 2-array, but we retain a little more type info. However while `basicEdge` is now the only `dag.Edge` implementation, we actually don't define if `Hashcode` must be the same or different for different edge types connecting the same nodes 🤔, so perhaps `[2]interface{}` is safer for now, until we finally get around to designing a better `dag` package 😁